src/Microsoft.Diagnostics.Monitoring.StartupHook/ParameterCapturing/MethodSignature.cs

```diff
@@ -58,6 +62,30 @@ private static string GetMethodName(MethodInfo method)
             return builder.ToString();
         }
 
+        private static int GetMethodToken(MethodInfo method)
+        {
+            try
+            {
+                return method.MetadataToken;
+            }
+            catch
```

Comment:
Do we actually get exceptions under any circumstances? If not, we should probably not wrap these with generic exception handling.

Reply:
See https://github.com/dotnet/dotnet-monitor/pull/7974#discussion_r1975922750

Reply:
I think comments in source would help alleviate someone from wondering why these are guarded.